### PR TITLE
Redux-components

### PR DESCRIPTION
**Pull request by AJ193**
### SET-UP REDUX STORE
- Configure a Redux store
- Set a slice of state for books that:
- Includes a reducer that checks the status and always returns "Under construction"

### CODE REVIEW

File: src/components/BookForm.jsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      <form action="" method="">
+        <input type="text" name="title" id="title" placeholder="Book Title" />
+        <input type="text" name="author" id="author" placeholder="Book Author" />
+        <button type="submit">Add Book</button>
+      </form>
+    </>
+  );
+}
+
+export default myForm;
```
Comment (Abenezer-Tilahun) [inline src/components/BookForm.jsx]: - [x] As stated in the project requirement Kindly make sure that you Add a <Button> component, which includes: An event handler **that adds a book to the books array (with attributes id, title and author)** Just like below code snippets:- 

```
import React from "react"
import { useDispatch } from "react-redux"
import { addBook } from "path/to/booksSlice"

function myForm() {
  const [title, setTitle] = useState("")
  const [author, setAuthor] = useState("")
  const books = useSelector((state) => state.books.books)
  const dispatch = useDispatch()

  // Other logic that controls form state

  return (
    <div>
      {/* Rest of form JSX */}
      <Button onClick={() => dispatch(addBook({ title, author }))}>
        Add book
      </Button>
    </div>
  )
}

export default myForm

```

- In the case of your project when i try to add a book to the book list , it doesn't work as shown with this [video](https://drive.google.com/file/d/1shZIpkRL-m6r9sLBNgt9EGy4V3qEL_gj/view?usp=sharing) Kindly Fix it.

File: src/components/myBook.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    </>
+  );
+}
+myBook.propTypes = {
+  id: PropTypes.number.isRequired,
+  title: PropTypes.string.isRequired,
+  author: PropTypes.string.isRequired,
+};
+
+export default myBook;
```
Comment (Abenezer-Tilahun) [inline src/components/myBook.js]: - [x] As stated in the project requirement Kindly make sure that you Add a <Button> component, which includes: **An event handler that removes a book from the books array (by id)** Just like the below code snippets :- 
```
import React from "react"
import { useDispatch } from "react-redux"
import { removeBook } from "path/to/booksSlice"

function myBook({ id, title, author }) {
  const dispatch = useDispatch()

  return (
    <div>
      {/* Rest of Book JSX */}
      <Button onClick={() => dispatch(removeBook(id))}>Remove</Button>
    </div>
  )
}

export default myBook
```

- But in the case of your project when i try to remove the book from the book list , it doesn't work as show with this [video](https://drive.google.com/file/d/1shZIpkRL-m6r9sLBNgt9EGy4V3qEL_gj/view?usp=sharing) So kindly fix it.

**Review by Abenezer-Tilahun — CHANGES_REQUESTED** [review]
Hi @AJ193   , 

Great job for implemented  the some of the Project Requirement , but there are some changes that need to be implemented in order to proceed to the next project 👍

## Highlights 👏🏻
✔️ No Linter errors
✔️ Correct Git-Flow
✔️ Good commit messages
✔️ Professional README

almost

## Required Changes ♻️
_Check the comments below the review._

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DuaneDave — APPROVED** [review]
Hi @AJ193 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Books can be added and remove, good job 👍 

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**